shells/web-shell/elements/web-shell.js

```diff
@@ -121,7 +121,8 @@ export class WebShell extends Xen.Debug(Xen.Async, log) {
     // setup environment once we have a root and a user
     if (!state.env && root) {
       state.env = this.configureEnv(root);
-      this.configureContext();
+      this.state = {runtime: state.env};
+      this.configureContext(state.env);
     }
```

Comment:
This was weird when I wrote it. Today I would do the below. Chef's choice.
```
const runtime = this.configureEnv(root);
this.state = {runtime, env: runtime};
this.configureContext(runtime);
```

Reply:
thanks for the suggestion, it's much nicer indeed! done